glossary.yml

```
@@ -4169,7 +4171,7 @@
   en:
     term: "prompt"
     def: >
-      The text printed by a [REPL](#repl) or [shell](#shell) that indicates it is
+      The text printed by an [REPL](#repl) or [shell](#shell) that indicates it is
```

Mostly curious: why use 'an' when the beginning of the acronym does not have a vowel sound?

```suggestion
      The text printed by a [REPL](#repl) or [shell](#shell) that indicates it is
```

---

I think it depends on whether you're pronouncing the acronym as a word or spelling it out: "a sequel database" versus "an ess cue ell database".

---

One of these days, my brain will work properly.... but probably not for a while.

---

@zkamvar, @gvwilson  Here is the comment I originally left regarding [a REPL/an REPL] and why the second is correct. It is a difference between the word (acronym, in this case) starting with a **vowel sound** versus starting with a **vowel**. (I think all words that start with a vowel also start with a vowel sound, because I can't think of any counterexamples; however, not all words that start with a vowel sound start with a vowel. I am uncertain as to the size of that list, though.)

#### Original comment
I understand why you would think this, but because the actual pronunciation of the letter 'R' starts with a vowel sound, it needs the 'n'.

(If you say R-E-P-L, the first syllable is the same as the first syllable of argument, and you say an argument.

---

@baileythegreen Thank you for the clarification. I did understand the difference between using an with vowel sounds, but never heard REPL spelled out before. 

This also brings up a good point that potential clashes between American and British rules (e.g. an herb garden vs. a herb garden) and it means that we either need to choose one flavor/flavour of English or forever be adding or subtracting single letter changes. 

There's a reason I love and hate the English language all the same. :weary:

---

@zkamvar I say this as an American who has lived in Britain for the last five years, and long time basher of English as a language: English is ridiculous. It makes no sense. But i think British English makes even less sense because they often randomly just don't pronounce letters.

I think it is possible that the 'herb' issue might be singular. I don't know of another word where the determination of whether it starts with a vowel sound versus not is dependent upon the speaker's accent. Luckily, 'herb' is unlikely to appear in this glossary.